### PR TITLE
Add support for selecting a single unit test

### PR DESCRIPTION
**Pull request by doru1004**
# Add support for single unit test selection

This PR is implementing a way to execute individual unit tests. Use the RAJA_PERF_UNIT_TEST environment variable to select the unit test to run. For example:
```
RAJA_PERF_UNIT_TEST=DAXPY ./build_omptarget/test/test-raja-perf-suite.exe
```

Companion PR for merging is: https://github.com/LLNL/RAJAPerf/pull/319. Please approve that PR.

### CODE REVIEW

File: test/test-raja-perf-suite.cpp [hunk 1]
```diff
@@ -32,27 +32,38 @@ TEST(ShortSuiteTest, Basic)
...
+
+  int next_index = 0;
+
   std::vector< std::string > sargv(argc);
-  sargv[0] = std::string("dummy ");  // for executable name
-  sargv[1] = std::string("--checkrun");
-  sargv[2] = std::string("3");
-  sargv[3] = std::string("--show-progress");
-  sargv[4] = std::string("--disable-warmup");
+  sargv[next_index++] = std::string("dummy ");  // for executable name
```
Comment (MrBurmark) [inline test/test-raja-perf-suite.cpp:42]: As long as we're thinking about changing this we should use c++ style `emplace_back` to add to `sargv` and let `argc = sargv.size()` at the end.

Comment (doru1004) [reply test/test-raja-perf-suite.cpp]: 👍

File: test/test-raja-perf-suite.cpp [hunk 2]
```diff
@@ -20,44 +20,39 @@ TEST(ShortSuiteTest, Basic)
...
-  argc = 7;
-#endif
-
-  std::vector< std::string > sargv(argc);
-  sargv[0] = std::string("dummy ");  // for executable name
-  sargv[1] = std::string("--checkrun");
-  sargv[2] = std::string("3");
-  sargv[3] = std::string("--show-progress");
-  sargv[4] = std::string("--disable-warmup");
+  std::vector< std::string > sargv = {};
```
Comment (MrBurmark) [inline test/test-raja-perf-suite.cpp:24]:
```suggestion
  std::vector< std::string > sargv{};
```

**Comment by rhornung67** [comment]
> @rhornung67 What do you think about this mechanism for selecting individual tests?

I think it's fine. However, I'm not sure why this is necessary for testing. Generally, we want to run all kernels when testing, except for the few cases where there are known failures for certain compilers/environments -- we already manually filter these out. I think it would be better to include a way to exclude kernels using https://github.com/LLNL/RAJAPerf/blob/develop/src/common/RunParams.cpp#L44. Then, we could use that on the command-line to filter out kernels in tests at run time rather than what we are currently doing https://github.com/LLNL/RAJAPerf/blob/develop/test/test-raja-perf-suite.cpp#L42 where it's baked into the source code.

**Comment by doru1004** [comment]
> > @rhornung67 What do you think about this mechanism for selecting individual tests?
> 
> I think it's fine. However, I'm not sure why this is necessary for testing. Generally, we want to run all kernels when testing, except for the few cases where there are known failures for certain compilers/environments -- we already manually filter these out. I think it would be better to include a way to exclude kernels using https://github.com/LLNL/RAJAPerf/blob/develop/src/common/RunParams.cpp#L44. Then, we could use that on the command-line to filter out kernels in tests at run time rather than what we are currently doing https://github.com/LLNL/RAJAPerf/blob/develop/test/test-raja-perf-suite.cpp#L42 where it's baked into the source code.

The idea is you don't know which kernels will fail ahead of time since we want to automate some of this testing.

**Comment by doru1004** [comment]
Any more thoughts on this patch?

**Comment by doru1004** [comment]
I see some test fails but I cannot see what the fail is. Please let me know if fixes are required on my side.

**Comment by rhornung67** [comment]
We've been having issues on our tioga machine related to software updates. It's not related to your contribution. We're working on it.

**Comment by doru1004** [comment]
Thanks for the update!

**Comment by rhornung67** [comment]
@doru1004 there's nothing to be done on this PR as we cannot merge a PR from a fork. The companion PR is here: https://github.com/LLNL/RAJAPerf/pull/319. It is approved and will be merged when CI checks pass.

**Comment by doru1004** [comment]
> @doru1004 there's nothing to be done on this PR as we cannot merge a PR from a fork. The companion PR is here: #319. It is approved and will be merged when CI checks pass.

Sounds great thanks so much! :)  I was looking at the wrong one!

**Comment by rhornung67** [comment]
Understand. It's a bit confusing. But, we have to do things this way due to LC security policies.

**Comment by doru1004** [comment]
Totally fine. We can just close this one:
Taken over by: https://github.com/LLNL/RAJAPerf/pull/319

**Pull request closed without merging**